### PR TITLE
git Initial vector of banks

### PR DESCRIPTION
**Pull request by matiaslavista**

### CODE REVIEW

**Comment by matiaslavista** [comment]
This is just an initial version, but the global vector of Particle::Bank and its internal primitives is visible from python

File: include/openmc/cppapi/bank.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef OPENMC_CPP_BANK_H
+#define OPENMC_CPP_BANK_H
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+//This file is exclusively to allow cppapi to see the declaration of adding
+//the bank to the pythong shared module
```
Comment (paulromano) [inline include/openmc/cppapi/bank.h]: typo: pythong

File: setup.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 
     # Data files and librarries
     'package_data': {
-        'openmc.cppapi':['openmc_cpp{}'.format(ext_suffix)],
+        'openmc.core':['openmc_cpp{}'.format(ext_suffix)],
```
Comment (paulromano) [inline setup.py]: Make we should call the library `openmc_core` instead of `openmc_cpp`

File: src/bank.cpp [hunk 3]
```diff
@@ -69,4 +69,24 @@ extern "C" int openmc_fission_bank(void** ptr, int64_t* n)
   }
 }
 
+//==============================================================================
+// CPP API
+//==============================================================================
+std::vector<Particle::Bank>& openmc_source_bank_cpp()
+{
+  if (simulation::source_bank.size() == 0) {
```
Comment (paulromano) [inline src/bank.cpp]: I prefer `!simulation::source_bank.empty()`

File: src/bank.cpp [hunk 4]
```diff
@@ -69,4 +69,24 @@ extern "C" int openmc_fission_bank(void** ptr, int64_t* n)
...
+    throw std::runtime_error("Source bank has not been allocated.");
+  }
+  return simulation::source_bank;
+}
+
+std::vector<Particle::Bank>& openmc_fission_bank_cpp()
+{
+  if (simulation::fission_bank.size() == 0) {
```
Comment (paulromano) [inline src/bank.cpp]: `!simulation::fission_bank.empty()`

File: src/bank.cpp [hunk 5]
```diff
@@ -69,4 +69,24 @@ extern "C" int openmc_fission_bank(void** ptr, int64_t* n)
   }
 }
 
+//==============================================================================
+// CPP API
+//==============================================================================
+std::vector<Particle::Bank>& openmc_source_bank_cpp()
```
Comment (paulromano) [inline src/bank.cpp]: We could just call this `source_bank()`. It's in the `openmc` namespace already so no need to prefix it with `openmc_`. The suffix also is not very meaningful.

File: src/bank.cpp [hunk 6]
```diff
@@ -69,4 +69,24 @@ extern "C" int openmc_fission_bank(void** ptr, int64_t* n)
...
+{
+  if (simulation::source_bank.size() == 0) {
+    throw std::runtime_error("Source bank has not been allocated.");
+  }
+  return simulation::source_bank;
+}
+
+std::vector<Particle::Bank>& openmc_fission_bank_cpp()
```
Comment (paulromano) [inline src/bank.cpp]: `fission_bank()`

File: src/cppapi/cppapi.cpp [hunk 7]
```diff
@@ -56,7 +57,12 @@ int next_batch()
 
 PYBIND11_MODULE(openmc_cpp, m) 
 {
+<<<<<<< HEAD:src/cppapi/cppapi.cpp
+  openmc::init_position(m);
+  openmc::init_bank(m);
+=======
```
Comment (paulromano) [inline src/cppapi/cppapi.cpp]: You have some leftover junk from a merge it looks like?

File: src/cppapi/position.cpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+//==============================================================================
+// CPP API
+//==============================================================================
+//Add elements of Position to the module
+void init_position(py::module& m)
+{
+  py::class_<Position> (m,"Position")
+    .def(py::init<double, double, double>())
+    .def(py::init<const vector<double>& >())
+    .def(py::init<const std::array<double, 3>&>())
```
Comment (paulromano) [inline src/cppapi/position.cpp]: Let's just keep the first constructor in the bindings (double, double, double). If a Python user wants to pass an iterable, they can unpack it with `*`.

File: src/cppapi/position.cpp [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+//==============================================================================
+// CPP API
+//==============================================================================
+//Add elements of Position to the module
+void init_position(py::module& m)
+{
+  py::class_<Position> (m,"Position")
+    .def(py::init<double, double, double>())
+    .def(py::init([](std::tuple<double,double,double> tup) {
```
Comment (matiaslavista) [inline src/cppapi/position.cpp:26]: Added this constructor for direct conversion from tuple. Needs to be expanded to implicitly transform ints

File: src/bank.cpp (hunk 5 above)
Comment (matiaslavista) [reply src/bank.cpp]: My only issue witht his is shadowing the simulation::source_bank, but I guess its retrieving it. I fixed it like you requested, but let me know if you want me to call the cpp side get_source_bank (Pythong could still get it like openmc.core.source_bank()

**Pull request merged**